server/src/user/user.service.ts

```diff
@@ -972,6 +1018,169 @@ export class UserService {
     await this.onetimeInviteRepository.delete(inviteId);
   }
 
+  async createOnetimeInviteRequest(
+    body: CreateOnetimeInviteRequestDto,
+    userId: number,
+  ) {
+    const {
+      invitingUserId: providedInvitingUserId,
+      communityId,
+      ...rest
+    } = body;
+
+    const user = await this.findOneOrFail(userId, ['communities']);
+    const isAdmin = user.admin;
+    let community: Community | undefined | null = user.communities.find(
+      (community) => community.id === communityId,
+    );
+    if (!community) {
+      if (!isAdmin) {
+        throw new UnauthorizedException(
+          `User is not a member of community ${communityId}`,
+        );
```

Comment:
this error message seems wrong/misleading, given that its thrown when the community doesnt exist?